examples/resnet-retrain/src/resnet_retrain/core.clj

```diff
@@ -259,7 +259,42 @@
     (* 4 batch-size vals-per-batch)))
 
 
-(defn -main
+(defn train-again
+  "incrementally improve upon the trained model"
   [& [batch-size]]
-  (train (when batch-size
-           (Integer/parseInt batch-size))))
+  (let [batch-size (or batch-size 32)
+        epoch-size 4096
+        network (util/read-nippy-file "trained-network.nippy")]
+    (println "training using batch size of" batch-size)
+    (train/train-n network
```

Comment:
This is what I'm not sure of - is it doing what I think it should be doing here, taking the RESNET50 network with frozen layers and just retraining the last layer?

Reply:
I am not super-familiar with this path, but it sounds like the answer to your question depends on the contents of "trained-network.nippy". You could inspect the layers in there and see if the ones you expect to be frozen, are (that is, they have `:learning-attenuation` zero).

hth.

Reply:
thanks - I'll check it out.

Reply:
Nice! Thank you too. This is an interesting area that can definitely use some attention so I'm glad you're here.

Reply:
yup - it is. The layers are all marked as `:non-trainable?`. The only ones that are trainable are the ones that are expected 
`({:node :labels, :non-trainable? nil} {:node :softmax-loss-1, :non-trainable? nil} {:node :fc21, :non-trainable? nil})`

Reply:
Hey, that's neat! Score one for models being just-clojure-data.